cpp/src/parquet/statistics.cc

```diff
@@ -552,12 +560,19 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
 
   void Merge(const TypedStatistics<DType>& other) override {
     this->num_values_ += other.num_values();
-    if (other.HasNullCount()) {
+    // Merge always runs when Merge builder's page statistics
+    // into column chunk statistics, so it tent to have null count.
+    if (ARROW_PREDICT_TRUE(other.HasNullCount())) {
       this->statistics_.null_count += other.null_count();
+    } else {
+      this->has_null_count_ = false;
     }
-    if (other.HasDistinctCount()) {
-      this->statistics_.distinct_count += other.distinct_count();
+    // Distinct count cannot be merged.
+    if (has_distinct_count_) {
+      has_distinct_count_ = false;
     }
+    // If !other.HasMinMax, might be all-nulls or nulls and nan,
```

Comment:
We cannot differentiate at least two states:
- (1) the stats has not seen any value except null or nan, so it does not have min/max
- (2) the stats has seen some valid values but min/max are truncated due to size limit

So if currently the stats is in the state (2), we should not merge min/max from the incoming stats

Reply:
The interesting thing is, currently, "merge" is only called by writer, and writer is always the case (1)...
@pitrou Do you have any idea here? Since `Statistics` is already public.

Reply:
I think we can keep the code as-is indeed.